recommend/policyRules.go

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Authors of KubeArmor
+
+package recommend
+
+import (
+	_ "embed" // need for embedding
+	"encoding/json"
+	"errors"
+
+	"github.com/fatih/color"
+	log "github.com/sirupsen/logrus"
+)
+
+// MatchSpec spec to match for defining policy
+type MatchSpec struct {
+	Name         string      `json:"name"`
+	Precondition string      `json:"precondition"`
+	Description  Description `json:"description"`
+	Rules        Rules       `json:"rules"`
+	OnEvent      OnEvent     `json:"onEvent"`
+}
+
+// Ref for the policy rules
+type Ref struct {
+	Name string   `json:"name"`
+	URL  []string `json:"url"`
+}
+
+// Description detailed description for the policy rule
+type Description struct {
+	Refs     []Ref  `json:"refs"`
+	Tldr     string `json:"tldr"`
+	Detailed string `json:"detailed"`
+}
+
+// PathRule specifics for the path/dir rule. Note that if the Path ends in "/" it is considered to be Directory rule
+type PathRule struct {
+	FromSource string   `json:"fromSource"`
+	Path       []string `json:"path"`
+	Recursive  bool     `json:"recursive"`
+}
+
+// Rules set of applicable rules. In the future, we might have other types of rules.
+type Rules struct {
+	PathRule PathRule `json:"pathRule"`
+}
+
+// OnEvent the information that is emitted in the telemetry/alert when the matching event is witnessed
+type OnEvent struct {
+	Severity int      `json:"severity"`
+	Message  string   `json:"message"`
+	Tags     []string `json:"tags"`
+	Action   string   `json:"action"`
+}
+
+var policyMatch []MatchSpec
+
+//go:embed json/rules.json
+var ruleSpecJSON []byte
```

Comment:
I am not sure if we should be embedding rules.json. The update frequency for this rule list will be pretty high and it wouldn't be feasible to release a new binary for each update to rules. We should perhaps host it at policy templates maybe and then  do a get request?

Reply:
We will pull the rules from the policy-template eventually like we had discussed. We do not want to keep a separate configuration file hence embedding the rules as part of the bin itself.

We will have two things done next (in subsequent PR):
1. have a way to pass an explicit rules.json in addition as an input from command line
2. update the rules.json from the policy-templates